### ISSUE TITLE
SpeechToText RecognizeStream stop() lag.

### ISSUE THREAD

**Issue by nevrologica** [watson-developer-cloud/speech-javascript-sdk#158]
Following the example for auto stopping a SpeechToText stream i noticed that there are cases where data received from the websocket can be identified as 'final' twice, that lead me to understand that stopping the stream on the first time a final data is received lags, causing an unexpected result in the chaining of events.

Sentence captured.
![text](https://user-images.githubusercontent.com/29121228/70896120-cbd05500-1ff8-11ea-8a51-488a49dd1170.png)

The code i am using.
![code](https://user-images.githubusercontent.com/29121228/70896157-dee32500-1ff8-11ea-9490-5b75fc0a92a4.png)

Console result.
![console](https://user-images.githubusercontent.com/29121228/70896181-ea365080-1ff8-11ea-8939-17fe1225fbc4.png)

Connections occured following the chaining of results.
![connections](https://user-images.githubusercontent.com/29121228/70896213-fcb08a00-1ff8-11ea-82c9-24a046fc6ee8.png)

And finally this is the part of the SDK code that handles the stop event.
![recognizestream](https://user-images.githubusercontent.com/29121228/70896248-0fc35a00-1ff9-11ea-8302-984bc817a0a0.png)

I understand that there can be numerous final results in a stream. This is actually there to handle the formation of the sentence by adding a '.' and begining a new sentence, if i understand it correctly. So the stream.stop() should stop the stream immediatly, also the connection, However my finding shows that whenever the stop event is emitted there are still data going and coming from the websocket just before it is closed causing the stream to fire the on.data event twice even if the stream has been stopped in a previous iteration.

I actually found out about this behaviour when i was tired speaking to the microphone for testing, and i was using a youtube video of an Elon Musk presentation with him speaking instead of my own voice. When i am testing i usually say something then stop. So after i stop speaking there is nothing coming in to the microphone and the first final data are causing one stream stop. However when i was using the video sounds of people speaking while pausing for some seconds, a second final result was coming in, that meant that the websocket send data when the stream.stop() was called, and it was still waiting for the return of results, causing a second final result to occur. I think one solution should be to when stop is emmited to forcefully break the connection before it has the time to send back results or clear the stream buffer before attaching the closingMessage.

**Comment by germanattanasio** [comment]
This is working as expected. Even if you stop sending data there could be a queue of data being recognized by the service and that's why you keep on getting results. It's the developer's responsibility to stop showing results or keep showing them until the service sends the last part.

What you can do is to remove the on data handler once you click on stop or have available to see if you should discard the messages or not.

**Comment by nevrologica** [comment]
Tried both methods you suggested.

1st - I set a flag to true the first time a final data is coming in thus preventing executing the code twice when a second final data is occuring from the queue. Noticed that the outputElement i set as parameter still receives then second final message, so to fix this i kept the text on the first iteration and on the next one i set the saved text on top of the final that included both. A bit crude solution, as i could see the outputElement text changing.

2nd - I prefered unbinding the data handler at the end of all the code i am executing, so no additional text is piped into the outputElement and i don't have to fiddle with flags and saving previous text in order to reset it.

Thank you for your suggestions, works like a charm now.
